src/test/scala/com/fulcrumgenomics/commons/collection/SelfClosingIteratorTest.scala

```diff
@@ -0,0 +1,30 @@
+package com.fulcrumgenomics.commons.collection
+
+import com.fulcrumgenomics.commons.util.UnitSpec
+
+class SelfClosingIteratorTest extends UnitSpec {
+  "SelfClosingIterator" should "return all the items and then call the close function" in {
```

Comment:
This test wont check that `() => closed` isn't called twice.  A better test would be to have
`var closed = 0` and `() => closed += 1`, and then check the value of `closed` (should be `1`).